src/encode.jl

```diff
@@ -0,0 +1,47 @@
+"""
+    encode(img; kwargs...)
+
+Encode image `img` into JPEG byte sequences.
+"""
+function encode(img::AbstractMatrix; transpose=false, kwargs...)
+    # quantilized into 8bit sequences first
+    AT = Array{n0f8(eltype(img)), ndims(img)}
+    # jpegturbo is a C library and assumes row-major memory order, thus `collect` the data into
+    # contiguous memeory layout already makes a transpose.
+    img = transpose ? convert(AT, img) : convert(AT, PermutedDimsArray(img, (2, 1)))
+
+    return _encode(img; kwargs...)
+end
+
+function _encode(img::Matrix{Gray{N0f8}}; quality=100)
+    cinfo = Ref{LibJpeg.jpeg_compress_struct}()
+    cinfo_ptr = Base.unsafe_convert(Ptr{LibJpeg.jpeg_compress_struct}, cinfo)
+    jerr = Ref{LibJpeg.jpeg_error_mgr}()
+    cinfo_ptr.err = LibJpeg.jpeg_std_error(jerr)
+    LibJpeg.jpeg_create_compress(cinfo_ptr)
+
+    cinfo_ptr.image_width = size(img, 2)
+    cinfo_ptr.image_height = size(img, 1)
+    cinfo_ptr.input_components = 1
+    cinfo_ptr.in_color_space = LibJpeg.JCS_GRAYSCALE
+
+    LibJpeg.jpeg_set_defaults(cinfo_ptr)
+    LibJpeg.jpeg_set_quality(cinfo_ptr, Cint(quality), btrue)
+
+    bufsize = Ref{Culong}(0)
+    buf_ptr = Ref{Ptr{UInt8}}(C_NULL)
+    LibJpeg.jpeg_mem_dest(cinfo_ptr, buf_ptr, bufsize)
+    LibJpeg.jpeg_start_compress(cinfo_ptr, btrue)
```

Comment:
```suggestion
    LibJpeg.jpeg_start_compress(cinfo, btrue)
```

You can directly pass `cinfo` without GC-preserving.